### PR TITLE
[NUI] Reduce code duplication - refactor dispose codes

### PR DESCRIPTION
**Pull request by dongsug-song**
### Description of Change ###
[NUI] Reduce code duplication - refactor dispose codes

Refactor dispose codes to reduce code duplication

### API Changes ###
 - ACR: http://suprem.sec.samsung.net/jira/browse/TCSACR-274

### CODE REVIEW

**Comment by WonyoungChoi** [comment]
please fix since_tizen 5.5 to 6

File: src/Tizen.NUI/src/internal/Disposable.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            {
+                Dispose(DisposeTypes.Explicit);
+                System.GC.SuppressFinalize(this);
+            }
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Disposable.cs]: It should be 6

File: src/Tizen.NUI/src/internal/Disposable.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            {
+                isDisposeQueued = true;
+                DisposeQueue.Instance.Add(this);
+            }
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Disposable.cs]: It should be 6

File: src/Tizen.NUI/src/internal/Disposable.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        public Disposable()
+        {
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Disposable.cs]: It should be 6

File: src/Tizen.NUI/src/internal/Disposable.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        /// A Flat to check if it is already disposed.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
+        protected bool disposed = false;
+        private bool isDisposeQueued = false;
+
+        /// <summary>
+        /// Create an instance of BaseHandle.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Disposable.cs]: It should be 6

File: src/Tizen.NUI/src/internal/Disposable.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    /// <summary>
+    /// Disposable class.
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
+    public class Disposable : global::System.IDisposable
+    {
+        /// <summary>
+        /// A Flat to check if it is already disposed.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Disposable.cs]: It should be 6

File: src/Tizen.NUI/src/internal/Disposable.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// Disposable class.
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Disposable.cs]: It should be 6

**Comment by dongsug-song** [comment]
@m-macias 
Hello, 
http://suprem.sec.samsung.net/jira/browse/TCSACR-274 has been accepted.
can I merge this?
please let me get your answer.
thank you.

**Comment by m-macias** [comment]
@dongsug-song  Please update the since_tizen version in the Disposable class, before you merge this pull request.
Thank you

**Pull request merged**